### PR TITLE
 Vet model structure and comment on code

### PR DESCRIPTION
**Pull request by jwalzberg**
Leaving as draft until review is over.

### CODE REVIEW

**Comment by jwalzberg** [comment]
@akey7 @rjhanes @TJTapajyoti: the vet_comment branch is ready for review and subsequent merge into the develop branch. 

High-level comments:
- This is an impressive model and I had a lot of fun getting to learn how celavi work! Also learned a couple of Python tricks so thank you 😊
- The code is a bit convoluted (there is a lot of back and forth between python modules) so at time it’s a bit hard to follow. However, I am not sure it’s really possible to organize better (and my codes are probably worse so I  can’t really say anything).
- I haven’t found any big issues. The “worse” I found are the few parts where I did not fully understand the logic/what was happening. I flagged those parts in my comments and suggested to clarify with comment lines.
- There is also a lot of minor comments regarding formatting and documenting the code. (For formatting, I know you will use the Black package, but it was sometime too frustrating to have the PyCharm suggestions and not do anything so I made a few formatting changes (I am used to coding while trying to keep the green tick on the top right of the PyCharm interface)).
- I also made suggestions where I thought the code could be better generalize (e.g., reusing general functions but with different inputs rather than having two almost similar functions).

Let me know if you have any questions.

File: celavi/pylca_celavi/pylca_celavi_background_postprocess.py [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# TODO: Add a short module docstring above the code to:
+#  1) provide authors, date of creation
+#  2) give a high level description (2-3 lines) of what the module does
+#  3) write any other relevant information
+
+# TODO: if sys, pickle, and os are not used, consider removing
```
Comment (rjhanes) [inline celavi/pylca_celavi/pylca_celavi_background_postprocess.py:6]: Leaving this one for TJ to review in detail

File: celavi/pylca_celavi/pylca_opt_background.py [hunk 2]
```diff
@@ -1,44 +1,77 @@
+# TODO: Add a short module docstring above the code to:
```
Comment (rjhanes) [inline celavi/pylca_celavi/pylca_opt_background.py:1]: This file was also extensively re-written in #101 - I believe all docstrings are now complete so I'll remove those TODO items and leave the rest for TJ.

File: celavi/pylca_celavi/pylca_opt_foreground.py [hunk 3]
```diff
@@ -7,6 +7,20 @@
 """
 
 
+# TODO: Consider generalizing "pylca_opt_foreground.py",
```
Comment (rjhanes) [inline celavi/pylca_celavi/pylca_opt_foreground.py:10]: I removed docstring TODOs and left the rest as above

File: celavi/routing.py [hunk 4]
```diff
@@ -23,20 +23,28 @@
...
         """
-        :param edges: [DataFrame]
-        :param node_map: [DataFrame]
-        :param memory [joblib.Memory]
-        :param algorithm: [function]
-        :return:
+
+        Parameters
+        ----------
+        edges: [DataFrame]
```
Comment (rjhanes) [inline celavi/routing.py:40]: @eberlea Can you fill out the init docstring for Router?

File: celavi/routing.py [hunk 5]
```diff
@@ -99,6 +116,10 @@ def get_route(self, start, end):
             .sum().reset_index()
 
         _summary['region_transportation'] = _summary['statefp'] + _summary['countyfp']
+        # TODO: Consider generalizing below. An input parameter should be used.
+        #  At least 0.621371 should be stored in a variable such as
+        #  mile_per_km (I suppose it's the conversion factor mile-km).
+        #  A comment should also specify why weight is divided by 1000.
```
Comment (rjhanes) [inline celavi/routing.py]: The mile-to-km conversion TODO item will be resolved with issue #108 as it also affects CostGraph code

File: celavi/routing.py [hunk 6]
```diff
@@ -99,6 +116,7 @@ def get_route(self, start, end):
             .sum().reset_index()
 
         _summary['region_transportation'] = _summary['statefp'] + _summary['countyfp']
+        # TODO:  A comment should also specify why weight is divided by 1000.
```
Comment (rjhanes) [inline celavi/routing.py:119]: @eberlea Can you add a comment on to address this TODO?

**Review by rjhanes — APPROVED** [review]
I've gone through the comments and TODO items and either resolved them with commits, deleted the TODO comment for anything that I know we won't do, or left a review comment with other information. I'm submitting my approval pending @TJTapajyoti and @akey7 's comments. Because there were quite a few issues identified that have since been addressed, it's probably a good idea to re-review the codebase once we have the next release prepared.

File: celavi/__main__.py [hunk 7]
```diff
@@ -188,8 +198,8 @@
 # files immediately.
 
 os.chdir(subfolder_dict['lci_folder'])
-from celavi.des import Context
-from celavi.diagnostic_viz import DiagnosticViz
+from celavi.des import Context  # TODO: is there no way this could be moved up?
+from celavi.diagnostic_viz import DiagnosticViz  # TODO: ditto above
```
[the remark this answers is not in the document]
Comment (TJTapajyoti) [reply celavi/__main__.py:209]: PR #101 is ready.

File: celavi/data_manager.py [hunk 8]
```diff
@@ -24,10 +24,25 @@ class Data(pd.DataFrame):
     INDEX_COLUMNS = []
 
     def __init__(self, df=None, fpath=None, columns=None, backfill=True):
+        """
```
[the remark this answers is not in the document]
Comment (TJTapajyoti) [reply celavi/data_manager.py:27]: I updated the data filtering docstrings in PR #101

File: celavi/pylca_celavi/concrete_life_cycle_inventory_editor.py [hunk 9]
```diff
@@ -1,3 +1,8 @@
+# TODO: Add a short module docstring above the code to:
+#  1) provide authors, date of creation
+#  2) give a high level description (2-3 lines) of what the module does
+#  3) write any other relevant information
+
```
[the remark this answers is not in the document]
Comment (TJTapajyoti) [reply celavi/pylca_celavi/concrete_life_cycle_inventory_editor.py:5]: Will be done in the next hackathon

File: celavi/pylca_celavi/des_interface.py [hunk 10]
```diff
@@ -1,3 +1,9 @@
+# TODO: Add a short module docstring above the code to:
+#  1) provide authors, date of creation
+#  2) give a high level description (2-3 lines) of what the module does
+#  3) write any other relevant information
+
+# TODO: if sys is not used, consider removing
```
[the remark this answers is not in the document]
Comment (TJTapajyoti) [reply celavi/pylca_celavi/des_interface.py]: pr #101  is ready

File: celavi/pylca_celavi/concrete_life_cycle_inventory_editor.py [hunk 11]
```diff
@@ -9,18 +14,29 @@
 """
 
 
-def concrete_life_cycle_inventory_updater(fd_cur2,yr,k,stage):
+def concrete_life_cycle_inventory_updater(fd_cur2, yr, k, stage):
+    # TODO: add docstrings to explain input variables and what the function
+    #  does.
+
+    # TODO: consider having sand_substitution_rate and coal_substitution_rate
```
Comment (TJTapajyoti) [inline celavi/pylca_celavi/concrete_life_cycle_inventory_editor.py:21]: The coal and sand substitution rate are being obtained from the config file. PR #101

File: celavi/pylca_celavi/pylca_celavi_background_postprocess.py (hunk 1 above)
Comment (TJTapajyoti) [reply celavi/pylca_celavi/pylca_celavi_background_postprocess.py:6]: Completed in PR #101

File: celavi/pylca_celavi/pylca_opt_foreground.py (hunk 3 above)
Comment (TJTapajyoti) [reply celavi/pylca_celavi/pylca_opt_foreground.py:10]: Done in PR # 101

File: celavi/pylca_celavi/pylca_opt_background.py (hunk 2 above)
Comment (TJTapajyoti) [reply celavi/pylca_celavi/pylca_opt_background.py:1]: Completed in PR #101

File: celavi/pylca_celavi/pylca_opt_background.py [hunk 12]
```diff
@@ -1,44 +1,75 @@
...
+#  1) provide authors, date of creation
+#  2) give a high level description (2-3 lines) of what the module does
+#  3) write any other relevant information
+
+# TODO: Consider using a class to organize the code below. Some variables
+#  currently defined outside functions (at places that may feel random to
+#  the reader) could then be defined in the __init__ function of the class.
+
```
Comment (TJTapajyoti) [inline celavi/pylca_celavi/pylca_opt_background.py:9]: There are lots of TODOs in this file. I will address them in another PR later.

File: celavi/pylca_celavi/pylca_opt_foreground.py [hunk 13]
```diff
@@ -7,6 +7,20 @@
...
+#  products and process lists, most of the pre-processing function etc. A
+#  class with more general function could be created and an instance of the
+#  class could be created when needed for the functions that compute the insitu
+#  emissions and foreground (part that are not similar between the two cases
+#  would be defined in those functions). A boolean, such as "insitu_emission",
+#  could be used for the code to differentiate inputs manipulations when needed
+#  (e.g., for the line "res.to_csv('intermediate_demand.csv', mode='a',
+#  header=False, index=False)" which is in pylca_opt_foreground.py, but not in
+#  insitu_emission.py or pylca_opt_background.py.
+
```
Comment (TJTapajyoti) [inline celavi/pylca_celavi/pylca_opt_foreground.py:22]: Lots of TODO in this file which ill be addressed in the next PR

File: celavi/des.py [hunk 14]
```diff
@@ -1,3 +1,4 @@
+# TODO: remove Union, Tuple and floor if they are not used
```
Comment (akey7) [inline celavi/des.py:1]: Removed unused imports at top of file.

File: celavi/component.py [hunk 15]
```diff
@@ -203,6 +210,11 @@ def eol_process(self, env):
                         env.now
                     )
 
+                    # TODO: consider adding a comment to explain why it is
```
Comment (akey7) [inline celavi/component.py]: Added a comment to the TransportationTracker class regarding why that class is necessary alongside FacilityInventories.

File: celavi/component.py [hunk 16]
```diff
@@ -218,10 +230,23 @@ def eol_process(self, env):
                         node_name = location,
                         connect_to = self.split_dict[factype]['facility_2'])
 
+                    # TODO: the contex.count_facility_inventorie,
```
Comment (akey7) [inline celavi/component.py]: Moved this comment to an issue to be completed later.

**Review by akey7 — APPROVED** [review]
Addressed issues in the Context and Component files.
